packages/cozy-stack-client/src/getIllegalCharacter.spec.js

```diff
@@ -0,0 +1,24 @@
+import { getIllegalCharacters } from './getIllegalCharacter'
+
+describe('getIllegalCharacters', () => {
+  it('should throw error if file name empty', () => {
+    expect.assertions(1)
+    const name = undefined
+    try {
+      getIllegalCharacters({ name })
+    } catch (error) {
+      expect(error.message).toEqual('file should have a name property ')
+    }
+  })
+
+  it('should return empty string when no illegal characters found', () => {
+    const name =
+      'Valid file name with #@&"\'()§è!çà)-°_^¨*$€ù%`£=+:;.,?∞…÷≠≤<>•ë“‘{¶«Çø}æê®†Úºîœπ‡Ò∂ƒﬁÌÏÈ¬µ‹≈©◊ß~ '
+    expect(getIllegalCharacters({ name })).toEqual('')
+  })
+
+  it('should return illegal characters split with spaces when illegal characters found', () => {
+    const name = 'Invalid file name with / and / and \x00 and \n and \r'
+    expect(getIllegalCharacters({ name })).toEqual('/ / \x00 \n \r')
```

Comment:
`\x00`, `\n` and `\r` won't be visible in the user interface, or will they?

Maybe we should escape them first?

Reply:
do you want me to add a backslash \ in front of them?

Reply:
let's resolve this

Reply:
Choice made after a call : `\x00`, `\n` and `\r` seem to be impossible to insert from the UI using classic way (i.e copy/paste correctly sanitizes input), so we think it would be useless to spend time to find a way to display them in the UI.

However the question may come back later if we add additional invisible forbidden characters.